### PR TITLE
upgrade dcrdata

### LINKED ISSUE #26: upgrade dcrdata by replacing the go mod to a specific commit
go mod replace to this commit https://github.com/decred/dcrdata/pull/1968

### PR DESCRIPTION
**Pull request by dreacot**
closes #26

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -34,7 +34,7 @@ require (
 	github.com/decred/dcrd/rpc/jsonrpc/types/v3 v3.0.0
 	github.com/decred/dcrd/txscript/v4 v4.1.0
 	github.com/decred/dcrd/wire v1.6.0
-	github.com/decred/dcrdata/v7 v7.0.0-20211216152310-365c9dc820eb
+	github.com/decred/dcrdata/v7 v7.0.0
```
Comment (dmigwi) [inline go.mod:37]: Is this really tied to the latest master commit? Why don't you tie it to this commit: https://github.com/decred/dcrdata/commit/e7e8ab2402609b0094a5b0ba05fa0a3dce67f053

Comment (dreacot) [reply go.mod:37]: no it's not, based on the issue https://github.com/crypto-power/cryptopower/issues/26 it's tied to the v1.8 commit

Comment (dmigwi) [reply go.mod:37]: My point is that the above dependency seems to be tied to the official release of v7.0.0 but the commit we want happened after that.

Comment (dmigwi) [reply go.mod:37]: Run this command to import changes as of this commit: https://github.com/decred/dcrdata/commit/fa4d8e1b4e8e2ea7752e5ce95c3c8e4a3beec6bd

```
 go get github.com/decred/dcrdata/v8@fa4d8e1b4e8e2ea7752e5ce95c3c8e4a3beec6bd
```

Comment (dreacot) [reply go.mod:37]: that was the exact command i executed

File: go.mod [hunk 2]
```diff
@@ -103,12 +102,13 @@ require (
 	github.com/decred/dcrd/lru v1.1.1 // indirect
 	github.com/decred/dcrd/rpc/jsonrpc/types/v4 v4.0.0 // indirect
 	github.com/decred/dcrd/txscript/v3 v3.0.0 // indirect
+	github.com/decred/dcrdata/v8 v8.0.0-20230617164141-fa4d8e1b4e8e // indirect
```
Comment (dreacot) [inline go.mod:105]: @dmigwi did you miss this?

File: go.mod (hunk 1 above)
Comment (dreacot) [reply go.mod]: @dmigwi some packages still made reference to v8. they have been upgraded

File: go.mod [hunk 3]
```diff
@@ -31,10 +31,10 @@ require (
 	github.com/decred/dcrd/dcrutil/v3 v3.0.0
 	github.com/decred/dcrd/dcrutil/v4 v4.0.1
 	github.com/decred/dcrd/hdkeychain/v3 v3.1.1
-	github.com/decred/dcrd/rpc/jsonrpc/types/v3 v3.0.0
+	github.com/decred/dcrd/rpc/jsonrpc/types/v4 v4.0.0
 	github.com/decred/dcrd/txscript/v4 v4.1.0
 	github.com/decred/dcrd/wire v1.6.0
-	github.com/decred/dcrdata/v7 v7.0.0-20211216152310-365c9dc820eb
+	github.com/decred/dcrdata/v8 v8.0.0-20230617164141-fa4d8e1b4e8e
```
Comment (crux25) [inline go.mod:37]: `v8.0.0-20230627030434-e7e8ab240260` this is the commit tied to the latest commit on master not `v8.0.0-20230617164141-fa4d8e1b4e8e`. Run delete the dcrdata module entry on go.mod and run `go mod tidy` it should resolve to the latest commit.

Comment (dreacot) [reply go.mod:37]: https://github.com/crypto-power/cryptopower/issues/26 we aren't trying to target the latest master commit

Comment (crux25) [reply go.mod:37]: Any specific reason why we are freezing on `fa4d8e1b4e8e`.

Comment (dreacot) [reply go.mod:37]: yes, because that is the v1.8 upgrade which we need, until an official release is done

Comment (crux25) [reply go.mod:37]: Makes sense. I see upstream have not made a release for some time.

**Review by crux25 — APPROVED** [review]

**Pull request merged**